clients/bmp_client.go

```diff
@@ -79,3 +79,181 @@ func (bc *bmpClient) SlPackages() (SlPackagesResponse, error) {
+	}
+
+	response := TaskOutputResponse{}
+
+	err = json.Unmarshal(responseBytes, &response)
+	if err != nil {
+		errorMessage := fmt.Sprintf("bmp: failed to decode JSON response, err message '%s'", err.Error())
+		return TaskOutputResponse{}, errors.New(errorMessage)
+	}
+
+	return response, nil
+}
+
+func (bc *bmpClient) UpdateStatus(serverId string, status string) (UpdateStatusResponse, error) {
+	path := fmt.Sprintf("%s/baremetal/%s/%s", bc.url, serverId, status)
+	responseBytes, errorCode, err := bc.httpClient.DoRawHttpRequest(path, "PUT", &bytes.Buffer{})
+	if err != nil {
+		errorMessage := fmt.Sprintf("bmp: could not calls /baremetal/'%s'/'%s' on BMP server, error message '%s'", serverId, status, err.Error())
+		return UpdateStatusResponse{}, errors.New(errorMessage)
+	}
+
+	if slcommon.IsHttpErrorCode(errorCode) {
+		errorMessage := fmt.Sprintf("bmp: could not call /baremetal/'%s'/'%s' on BMP server, HTTP error code: '%d'", serverId, status, errorCode)
+		return UpdateStatusResponse{}, errors.New(errorMessage)
+	}
+
+	response := UpdateStatusResponse{}
+
+	err = json.Unmarshal(responseBytes, &response)
+	if err != nil {
+		errorMessage := fmt.Sprintf("bmp: failed to decode JSON response, err message '%s'", err.Error())
+		return UpdateStatusResponse{}, errors.New(errorMessage)
+	}
+
+	return response, nil
+}
+
+func (bc *bmpClient) Login(username string, password string) (LoginResponse, error) {
+	path := fmt.Sprintf("%s/login/%s/%s", bc.url, username, password)
+	responseBytes, errorCode, err := bc.httpClient.DoRawHttpRequest(path, "GET", &bytes.Buffer{})
+	if err != nil {
+		errorMessage := fmt.Sprintf("bmp: could not calls /login/'%s'/'%s' on BMP server, error message '%s'", username, password, err.Error())
+		return LoginResponse{}, errors.New(errorMessage)
+	}
+
+	if slcommon.IsHttpErrorCode(errorCode) {
+		errorMessage := fmt.Sprintf("bmp: could not call /login/'%s'/'%s' on BMP server, HTTP error code: '%d'", username, password, errorCode)
+		return LoginResponse{}, errors.New(errorMessage)
+	}
+
+	response := LoginResponse{}
+
+	err = json.Unmarshal(responseBytes, &response)
+	if err != nil {
+		errorMessage := fmt.Sprintf("bmp: failed to decode JSON response, err message '%s'", err.Error())
+		return LoginResponse{}, errors.New(errorMessage)
+	}
+
+	return response, nil
+}
+
+func (bc *bmpClient) CreateBaremetal(createBaremetalInfo CreateBaremetalInfo) (CreateBaremetalResponse, error) {
+	createBaremetalParameters := CreateBaremetalParameters{
+		Parameters: createBaremetalInfo,
+	}
+
+	requestBody, err := json.Marshal(createBaremetalParameters)
+	if err != nil {
+		errorMessage := fmt.Sprintf("bmp: failed to encode Json File, error message '%s'", err.Error())
+		return CreateBaremetalResponse{}, errors.New(errorMessage)
+	}
+
+	path := fmt.Sprintf("%s/%s", bc.url, "sl/packages")
+	responseBytes, errorCode, err := bc.httpClient.DoRawHttpRequest(path, "Post", bytes.NewBuffer(requestBody))
```

Comment:
I am pretty sure this should be `"POST"` not `"Post"` also for consistency reasons.